Created Play method. Now all the question instances are created in th…

paulaOliverosHollis opened the pull request:
…e ComputerGeekGame constructor.

Created a list for each of the types of questions as well as methods which randomize the order or the questions every time Play() is called.

The score and time are both initiated at the beginning of Play() and they are calculate and printed to the screen when the game is over.

The method in charge of randomizing the answer options is now up and running and seems to be working great.

Changed some small formatting issues that make the game look better overall.

Commit:
Created Play method. Now all the question instances are created in the ComputerGeekGame constructor.

Created a list for each of the types of questions as well as methods which randomize the order or the questions every time Play() is called.

The score and time are both initiated at the beginning of Play() and they are calculate and printed to the screen when the game is over.

The method in charge of randomizing the answer options is now up and running and seems to be working great.

Changed some small formatting issues that make the game look better overall.

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -2,12 +2,15 @@
 using System.Collections.Generic;
 using System.IO;
 
+
```
Whitespace

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -136,7 +153,42 @@ private bool UserChoseToQuit(char userMenuChoice)
 
```
A nice to have would be to update your menu functions to use enums instead of passing chars, but since we are trying to get this out the door, that is optional.

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -136,7 +153,42 @@ private bool UserChoseToQuit(char userMenuChoice)
+            RandomizeTrueOrFalseQuestions();
+
+            // Sort answer options in random order every time call Play()
+            foreach(var question in _multipleChoiceQuestions)
```
I would put this loop in the RandomizeMultipleChoiceQuestions method, as this seems like it should be part of that

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -136,7 +153,42 @@ private bool UserChoseToQuit(char userMenuChoice)
+                }
+            }
+
+            DateTime finishTime = DateTime.Now;
```
Why create a finish variable if you immediately throw it away. I would just say:
``` c#
TimeSpan completionTime = DateTime.Now - startTime;
```

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -148,5 +200,97 @@ private void PrintLeaderBoard()
+            List<MultipleChoiceQuestion> tempQuestionList = new List<MultipleChoiceQuestion>();
+            Random random = new Random();
+
+            tempQuestionList.AddRange(_multipleChoiceQuestions);
```
Remind me later to tell you about the most performant way to do this, but also why I think your solution is fine.

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -148,5 +200,97 @@ private void PrintLeaderBoard()
+        private void RandomizeMultipleChoiceQuestions()
+        {
+            List<MultipleChoiceQuestion> tempQuestionList = new List<MultipleChoiceQuestion>();
+            Random random = new Random();
```
Since you use this 2x every time the play method is called (Here and in RandomizeTrueFalse), I would create a field to store this. This would also fix the issue of overseeding your random class (you should only seed your random class once so it can work as well as possible)

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -148,5 +200,97 @@ private void PrintLeaderBoard()
+
+            while (tempQuestionList.Count > 0)
+            {
+                int randomIndex = random.Next(0, tempQuestionList.Count - 1);
```
![image](https://user-images.githubusercontent.com/43797367/63059441-6e4ac100-bebd-11e9-8dfb-f283cc5ac6bf.png)

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -148,5 +200,97 @@ private void PrintLeaderBoard()
+
+            while (tempQuestionList.Count > 0)
+            {
+                int randomIndex = random.Next(0, tempQuestionList.Count - 1);
```
![image](https://user-images.githubusercontent.com/43797367/63059446-7440a200-bebd-11e9-824a-991b6dc9ae23.png)

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -148,5 +200,97 @@ private void PrintLeaderBoard()
+        private MultipleChoiceQuestion.Answers GetMcAnswerFromPlayer()
+        {               
+
+            while (true)
```
Whitespace

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -148,5 +200,97 @@ private void PrintLeaderBoard()
+                Console.Write("\n\tAnswer: ");
+                string answerStg = Console.ReadLine();
+
+                if(answerStg.Length < 1 || string.IsNullOrWhiteSpace(answerStg))
```
This whole while loop is very off, you only tell them invalid input every other answer, it should be everytime

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -148,5 +200,97 @@ private void PrintLeaderBoard()
+            }
+
+            
+        }
```
Whitespace

zhollis21 commented on DeveloperGame/DeveloperGame/ComputerGeekGame.cs:
```diff
@@ -148,5 +200,97 @@ private void PrintLeaderBoard()
+                string answerStg = Console.ReadLine();
+
+
+                if (answerStg.Length < 1 || string.IsNullOrWhiteSpace(answerStg))
```
Whitespace, also same issue with the loop, it seems like your tried to do 2 different ways at the same time. You tell them invalid answer every other time

zhollis21 commented on DeveloperGame/DeveloperGame/MultipleChoiceQuestion.cs:
```diff
@@ -23,15 +23,33 @@ public override void PrintQuestion()
             base.PrintQuestion();
 
-            foreach (string answer in _answerOptions)
+            for(int i = 0; i < 4; i++)
```
You need to do a Control+K+D on this file, the spacing is off

zhollis21 commented on DeveloperGame/DeveloperGame/MultipleChoiceQuestion.cs:
```diff
@@ -23,15 +23,33 @@ public override void PrintQuestion()
+            for(int i = 0; i < 4; i++)
             {
-                Console.WriteLine(answer);
+                Console.WriteLine((Answers)i + $". {_answerOptions[i]}");
```
That's a cool way to use your enum

zhollis21 commented on DeveloperGame/DeveloperGame/TrueOrFalseQuestion.cs:
```diff
@@ -15,9 +15,7 @@ public TrueOrFalseQuestion(string question, bool correctAnswer) : base(question)
-            base.PrintQuestion();
-
-            Console.Write("\n\tTrue or False: ");
+            base.PrintQuestion();            
```
Why did you take out the True or False printing?

zhollis21 commented on DeveloperGame/DeveloperGame/MultipleChoiceQuestion.cs:
```diff
@@ -23,15 +23,33 @@ public override void PrintQuestion()
+
+        public void RandomizeAnswers()
+        {            
+            Random random = new Random();
```
Create a random field so you don't new it up everytime

zhollis21 commented on DeveloperGame/DeveloperGame/MultipleChoiceQuestion.cs:
```diff
@@ -23,15 +23,33 @@ public override void PrintQuestion()
+
+            _answerOptions.Add(temp);
+
+            _correctAnswer = (Answers)randomIndex;
```
This is a fairly good algorithm but I have 2 complaints:

1. Leave some comments so it's easier to follow along, I had to draw it out on a napkin to understand what is going on
2. The correct answer can never be D which seems like an issue

zhollis21 reviewed (changes requested):
I know you want to get this done, but please be more observant, it was a significant decrease in code cleanliness and functionality.

Commit:
I fixed several issues from pull request review.

Created UpdateLeaderBoard method as well as an updateFile method.  Game seems to be working with no problems.

Commit:
FIxed issues in Ramdomize methods.

Commit:
Made a few final changes to the RandomizeAnswers method.

zhollis21 commented on DeveloperGame/DeveloperGame/MultipleChoiceQuestion.cs:
```diff
@@ -10,6 +10,7 @@ public enum Answers { A, B, C, D }
 
         private Answers _correctAnswer;
         private List<string> _answerOptions;
+        private Random random = new Random();
```
```suggestion
        private Random _random = new Random();
```

zhollis21 commented on DeveloperGame/DeveloperGame/MultipleChoiceQuestion.cs:
```diff
@@ -23,15 +24,32 @@ public override void PrintQuestion()
+            {
+                // Indexes we want to swap
+                int correctIndex = (int)_correctAnswer;
+                int randomIndex = random.Next(0, 4);
```
```suggestion
                int randomIndex = _random.Next(0, 4);
```

zhollis21 committed:
Update DeveloperGame/DeveloperGame/MultipleChoiceQuestion.cs

zhollis21 committed:
Update DeveloperGame/DeveloperGame/MultipleChoiceQuestion.cs

zhollis21 reviewed (approved):
Awesome Possum!
